sdk/src/quotes/public/increase-liquidity-quote.ts

```diff
@@ -101,6 +104,37 @@ export function increaseLiquidityQuoteByInputTokenWithParams(
     `input token mint ${param.inputTokenMint.toBase58()} does not match any tokens in the provided pool.`
   );
 
+  const {
+    adjustedSqrtPrice: lowerAdjustedSqrtPrice,
+    adjustedTickCurrentIndex: lowerAdjustedTickCurrentIndex,
+  } = adjustPriceForSlippage(param.sqrtPrice, param.slippageTolerance, true);
+  const {
+    adjustedSqrtPrice: upperAdjustedSqrtPrice,
+    adjustedTickCurrentIndex: upperAdjustedTickCurrentIndex,
+  } = adjustPriceForSlippage(param.sqrtPrice, param.slippageTolerance, false);
+
+  const lowerQuote = getQuoteByInputToken({
+    ...param,
+    sqrtPrice: lowerAdjustedSqrtPrice,
+    tickCurrentIndex: lowerAdjustedTickCurrentIndex,
+  });
+  const quote = getQuoteByInputToken(param);
+  const upperQuote = getQuoteByInputToken({
+    ...param,
+    sqrtPrice: upperAdjustedSqrtPrice,
+    tickCurrentIndex: upperAdjustedTickCurrentIndex,
+  });
+
+  return {
+    tokenMaxA: BN.max(BN.max(lowerQuote.tokenEstA, upperQuote.tokenEstA), quote.tokenEstA),
+    tokenMaxB: BN.max(BN.max(lowerQuote.tokenEstB, upperQuote.tokenEstB), quote.tokenEstB),
+    tokenEstA: quote.tokenEstA,
+    tokenEstB: quote.tokenEstB,
+    liquidityAmount: quote.liquidityAmount,
```

Comment:
in trade context, "slippage" is used to fail transaction if the amount is beyond the threshold.
And it will work to protect users from sandwitch atack.

I think in the following case tokenEstA will be larger than required and deposit will succeed even if the price move to lower than slippage applied price.

- position range [92, 94] USDC/SOL
- current price: 93
- lowerAdjustedPrice: 93 x 99 = 92.07
- upperAdjustedPrice: 93 x 101 = 93.93
- input token: USDC
- input amount: 100 USDC

lowerQuote.liquidity will be calculated to deposit 100 USDC between [92, 92.07]. It is tight and results higher liquidity.
Higher liquidity will require more SOL distributed to 92.07 ~ 94.
The SOL amount will be larger than required SOL amount at 92.07 with quote.liquidity (smaller than lowerQuote.liquidity)

Since the Whirlpool contract requires a fixed liquidity input, I think we need to have maxA and maxB as the minimum amount of tokens that can achieve "liquidity" even if price moved between [current price - slippage, current price + slippage], WDYT?

Reply:
The new function `increaseLiquidityQuoteByInputTokenUsingPriceSlippage`'s only job now is to infer liquidity from the input token amount and then call `increaseLiquidityQuoteByLiquidityWithParams`.

`increaseLiquidityQuoteByLiquidityWithParams` handles the slippage calculation as well. 

Each new quote function now has it's own set of test suite to verify tokenEst & tokenMax values.